### PR TITLE
Update to support ExtJS5

### PR DESCRIPTION
**Pull request by wombleton**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,6 +67,7 @@ The best place to ask for help is on the [Deft JS Google Group](https://groups.g
 
 # Version History
 
+* 0.10.0 - ExtJS 5 compliance; removed Controllable.
```
Comment (superstructor) [inline README.md]: version should be 5.0.0

File: packages/deft/build.properties [hunk 2]
```diff
@@ -1,7 +1,7 @@
-deftjs.version=0.9.1
+deftjs.version=0.10.1
```
Comment (superstructor) [inline packages/deft/build.properties]: version should be 5.0.0

File: packages/deft/jsduck.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-  "--title": "Deft JS - v0.9.1 API Documentation",
+  "--title": "Deft JS - v0.10.1 API Documentation",
```
Comment (superstructor) [inline packages/deft/jsduck.json]: version should be 5.0.0

File: packages/deft/src/coffee/ioc/Injector.coffee [hunk 4]
```diff
@@ -233,7 +233,7 @@ Ext.define( 'Deft.ioc.Injector',
 				setterFunctionName = 'set' + Ext.String.capitalize( name )
 				targetInstance[ setterFunctionName ].call( targetInstance, value )
 		else
-			if Ext.getVersion( 'extjs' )? and targetInstance instanceof Ext.ClassManager.get( 'Ext.Component' )
+			if targetInstance instanceof Ext.ClassManager.get( 'Ext.Component' )
 				# NOTE: Ext JS's Ext.Component doesn't "play by the rules" and never actually calls Ext.Base::initConfig().
```
Comment (superstructor) [inline packages/deft/src/coffee/ioc/Injector.coffee]: In Ext JS 5.x Ext.Component now calls initConfig.

File: packages/deft/src/coffee/mvc/ViewController.coffee [hunk 5]
```diff
@@ -124,25 +124,25 @@ Ext.define( 'Deft.mvc.ViewController',
 		'Deft.mvc.ComponentSelector'
```
Comment (superstructor) [inline packages/deft/src/coffee/mvc/ViewController.coffee]: Delete all mvc / ViewController related code completely. Ext JS 5.x has ViewControllers etc.

File: packages/deft/test/karma/ext/5.0.0.conf.js [hunk 6]
```diff
@@ -34,7 +34,6 @@ module.exports = function(config) {
 			'test/js/log/Logger.js',
 			'test/js/ioc/Injector.js',
 			'test/js/mixin/Injectable.js',
-			'test/js/mixin/Controllable.js',
 			'test/js/mvc/ViewController.js',
```
Comment (superstructor) [inline packages/deft/test/karma/ext/5.0.0.conf.js]: Remove ViewController.js from Karma config.

File: packages/deft/test/karma/ext/5.0.0.conf.js [hunk 7]
```diff
@@ -52,7 +51,7 @@ module.exports = function(config) {
 
 		coverageReporter: {
 			type: 'html',
-			dir: 'test/coverage/ext/4.2.0'
+			dir: 'test/coverage/ext/5.0.0'
```
Comment (superstructor) [inline packages/deft/test/karma/ext/5.0.0.conf.js:53]: Where is the touch 5.x karma test config ?

**Comment by brian428** [comment]
A bit late here, but just a note that I'm thinking we'll still have a Deft ViewController that extends the Sencha ViewController. Looks like we may not need to deal with the control: block for view refs and events, but things like the automatic use of the injectable and observer mixins would still be needed. 

It might also make sense to help ease the transition to Ext 5. If our VC extends Sencha's VC, we can try to set things up such that folks' existing VCs will work with no (or minimal) changes in Ext 5?

**Comment by johnyanarella** [comment]
I'd raise that question in the discussion group, if not for the NDA.

I agree that we should discuss this a bit more before we dump our ViewControllers.

On the one hand, I don't want us to have to maintain features that duplicate existing functionality that is part of the standard framework.  On the other hand, I'd hate for everyone to have to spend hours rewriting their VCs in order to upgrade.  I've seen tweets from Sencha people promoting that upgrading to Ext JS 5 was a 30 minute effort.  I'd hate for Deft JS applications to become known for being harder to make that transition than a vanilla Ext JS application.

Our "companion view controller" feature might also still be relevant as a feature of a Deft ViewController that extends the Sencha ViewController.

We should evaluate how refs and handlers are defined in the new Sencha ViewControllers vs Deft JS ViewControllers.  What are the relative strengths and weaknesses of each?  Are Deft JS users losing anything if we eliminate our control block feature and associated functionality (such as live listeners)?  Can we provide an easier migration path?  For example, could we provide a helper function (extracted to an optional utility class) that parses current Deft JS control blocks and equivalently configures a new Sencha ViewController?

**Comment by brian428** [comment]
Exactly. First, I think there are definitely some features of Deft VCs that
we know Sencha isn't going to incorporate (observe, companions, inject,
etc.). Second, John's thoughts on easing the migration to Ext 5 are what I
was also thinking. Maybe we don't even need a helper function...maybe we
can add logic that takes an existing Deft JS control block and
automatically converts it into the equivalent Sencha VC control block, so
the user doesn't even have to do anything else.

If making it do this automatically is a problem (e.g. how do we know they
want their control block translated), perhaps a new VC annotation
("convertDeftControl: true") is one option. Or maybe it's time for a proper
Deft.config() method, to allow setting a compatibility flag, logging
options, and so on?

**Comment by superstructor** [comment]
Thanks for the good feedback. We can certainly revisit this prior to a 5.x release. For now to keep things moving quickly we just wanted to get all the tests for injection and promises passing, and also incorporate Brian's mixin conversion branch which is our next priority.

**Pull request merged**